File: boilerplate.hs
{-# LANGUAGE OverloadedStrings #-}
 
import System.IO
import System.Environment  
import System.Directory
import Control.Monad
import Network.HTTP.Conduit
import qualified Data.ByteString.Lazy as L
import qualified Data.ByteString.Char8 as C
 
(email, name) = ("", "")
 
main = do
  args <- getArgs
  mapM_ putStrLn args
  dir <- getCurrentDirectory
  putStrLn dir
  initReq <- parseUrl "http://mipt.eu01.aws.af.cm/lab1"
  handle <- openFile (dir ++ "/boilerplate.hs") ReadMode
  content <- hGetContents handle
  let req = urlEncodedBody [("email", email), ("name", name), ("content", C.pack content)] $ initReq { method = "POST" }
  response <- withManager $ httpLbs req
  L.putStr $ responseBody response

File: boilerplate.fs
open System
open System.Net
open System.IO
open System.Collections.Specialized

let (email, name) = ("", "")

let values = new NameValueCollection()
values.Add("email", email)
values.Add("name", name)
values.Add("content", File.ReadAllText(__SOURCE_DIRECTORY__ + @"\" + __SOURCE_FILE__))

let client = new WebClient()
let response = client.UploadValues(new Uri("http://mipt.eu01.aws.af.cm/lab1"), values)
let responseString = Text.Encoding.Default.GetString(response)

printf "%A" responseString
